src/ClientData/include/ClientData/FunctionInfo.h

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CLIENT_DATA_FUNCTION_INFO_H_
+#define CLIENT_DATA_FUNCTION_INFO_H_
+
+#include <optional>
+#include <string>
+#include <utility>
+
+#include "GrpcProtos/symbol.pb.h"
+
+namespace orbit_client_data {
+
+class ModuleData;
+class ProcessData;
+
+// This class is used on the client to represent a function, containing a name, module path and
+// build ID, address and size.
+class FunctionInfo {
+ public:
+  FunctionInfo() = delete;
+  FunctionInfo(std::string pretty_name, std::string module_path, std::string module_build_id,
+               uint64_t address, uint64_t size)
+      : pretty_name_(std::move(pretty_name)),
+        module_path_(std::move(module_path)),
+        module_build_id_(std::move(module_build_id)),
+        address_(address),
+        size_(size) {}
+
+  FunctionInfo(const orbit_grpc_protos::SymbolInfo& symbol_info, std::string module_path,
+               std::string module_build_id)
+      : pretty_name_{symbol_info.demangled_name()},
+        module_path_{std::move(module_path)},
+        module_build_id_{std::move(module_build_id)},
+        address_{symbol_info.address()},
+        size_{symbol_info.size()} {}
+
+  [[nodiscard]] const std::string& pretty_name() const { return pretty_name_; }
+  [[nodiscard]] const std::string& module_path() const { return module_path_; }
+  [[nodiscard]] const std::string& module_build_id() const { return module_build_id_; }
+  [[nodiscard]] uint64_t address() const { return address_; }
+  [[nodiscard]] uint64_t size() const { return size_; }
+
+  [[nodiscard]] std::string GetLoadedModuleName() const;
+  [[nodiscard]] uint64_t GetHash() const;
```

Comment:
This function name could be confusing because it doesn't become clear that this is some special Hash for the pretty name. Maybe `GetPrettyNameHash`?